arrow-libs/optics/arrow-optics-ksp-plugin/src/main/kotlin/arrow/optics/plugin/internals/domain.kt

```diff
@@ -1,14 +1,21 @@
 package arrow.optics.plugin.internals
 
+import arrow.optics.plugin.companionObject
+import com.google.devtools.ksp.getVisibility
 import com.google.devtools.ksp.symbol.KSClassDeclaration
 import com.google.devtools.ksp.symbol.KSName
+import com.google.devtools.ksp.symbol.Visibility
 import java.util.Locale
 
 data class ADT(val pckg: KSName, val declaration: KSClassDeclaration, val targets: List<Target>) {
   val sourceClassName = declaration.qualifiedNameOrSimpleName
   val sourceName = declaration.simpleName.asString().replaceFirstChar { it.lowercase(Locale.getDefault()) }
   val simpleName = declaration.simpleName.asString()
   val packageName = pckg.asString()
+  val visibilityModifierName = when (declaration.companionObject?.getVisibility()) {
```

Comment:
Why handle only `INTERNAL` and not the rest of visibility modifiers?

Reply:
- `private` we can not handle here, because we cannot generate a `private` function for a `private` value of another file.

- `protected` doesn't make sense either because it's only allowed in combination with inheritance, so it's also not accessible. (Right?)